logrotate/test/RotateFileTest.cpp

```diff
@@ -1,304 +1,325 @@
+  ASSERT_FALSE(checkIfExist("msg4")) << "\n\tcontent:" << content;
 }
 
-
 TEST_F(RotateFileTest, setFlushPolicy__force_flush) {
-   LogRotate logrotate(_filename, _directory);
-   auto logfilename = logrotate.logFileName();
-   logrotate.setFlushPolicy(100);
-
-   std::string content;
-   auto checkIfExist = [&](std::string expected) -> bool {
-      content = ReadContent(logfilename);
-      bool exists = Exists(content, expected);
-      return exists;
-   };
-
-   // auto buffer size if by default 1024
-   logrotate.save("msg1\n");
-   ASSERT_FALSE(checkIfExist("msg1")) << "\n\tcontent:" << content;
-
-   logrotate.save("msg2\n");
-   ASSERT_FALSE(checkIfExist("msg2")) << "\n\tcontent:" << content;
-
-   logrotate.save("msg3\n");
-   logrotate.flush();
-   ASSERT_TRUE(checkIfExist("msg3")) << "\n\tcontent:" << content;   // 3rd write flushes it + previous
-
-   logrotate.save("msg4\n");
-   logrotate.flush();
-   ASSERT_TRUE(checkIfExist("msg4")) << "\n\tcontent:" << content;   // 3rd write flushes it + previous
+  LogRotate logrotate(_filename, _directory);
+  auto logfilename = logrotate.logFileName();
+  logrotate.setFlushPolicy(100);
+
+  std::string content;
+  auto checkIfExist = [&](std::string expected) -> bool {
+    content = ReadContent(logfilename);
+    bool exists = Exists(content, expected);
+    return exists;
+  };
+
+  // auto buffer size if by default 1024
+  logrotate.save("msg1\n");
+  ASSERT_FALSE(checkIfExist("msg1")) << "\n\tcontent:" << content;
+
+  logrotate.save("msg2\n");
+  ASSERT_FALSE(checkIfExist("msg2")) << "\n\tcontent:" << content;
+
+  logrotate.save("msg3\n");
+  logrotate.flush();
+  ASSERT_TRUE(checkIfExist("msg3"))
+      << "\n\tcontent:" << content;  // 3rd write flushes it + previous
+
+  logrotate.save("msg4\n");
+  logrotate.flush();
+  ASSERT_TRUE(checkIfExist("msg4"))
+      << "\n\tcontent:" << content;  // 3rd write flushes it + previous
 }
 
-
-TEST_F(RotateFileTest, DISABLED_setMaxArchiveLogCount) {
-   EXPECT_FALSE(true);
-}
+TEST_F(RotateFileTest, DISABLED_setMaxArchiveLogCount) { EXPECT_FALSE(true); }
+
+TEST_F(RotateFileTest, rotateLog) {
+  LogRotate logrotate(_filename, _directory);
+  std::string logfilename = logrotate.logFileName();
+  std::cout << logfilename << std::endl;
+  std::string content;
+  auto checkIfExist = [&](std::string expected) -> bool {
+    content = ReadContent(logfilename);
+    bool exists = Exists(content, expected);
+    return exists;
+  };
+  logrotate.save("test1");
+  ASSERT_TRUE(checkIfExist("test1")) << "\n\tcontent:" << content;
+  EXPECT_TRUE(DoesFileEntityExist(logfilename));
+  logrotate.rotateLog();
+  logrotate.save("test2");
+  ASSERT_FALSE(checkIfExist("test1")) << "\n\tcontent:" << content;
+  ASSERT_TRUE(checkIfExist("test2")) << "\n\tcontent:" << content;
+  auto app_name = _filename + ".log";
+  logrotate.rotateLog();
+  logrotate.save("test3");
+  auto allFiles =
+      LogRotateUtility::getLogFilesInDirectory(_directory, app_name);
```

Comment:
This looks good. We aren't testing the number of gz files existence but **I think that's fine.** 
I.e. there is no need to look into the benign time-race condition where  the `test1` `gz` file will be overwritten by the `test1`  - unless a new second period has started.